codalab/worker/docker_image_manager.py

```diff
@@ -243,10 +243,21 @@ def image_availability_state(image_spec, success_message, failure_message):
                 def download():
                     logger.debug('Downloading Docker image %s', image_spec)
                     try:
-                        self._docker.images.pull(image_spec)
+                        client = docker.APIClient(base_url='unix://var/run/docker.sock')
+                        self.generator = client.pull(image_spec, stream=True, decode=True)
+                        for line in self.generator:
+                            self._downloading[image_spec]['status'] = (
```

Comment:
I think we can pull this line out of the for loop

Reply:
how do you mean?

Reply:
Oh, I see what you were doing. Never mind my comment